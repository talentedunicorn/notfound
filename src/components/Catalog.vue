<template>
  <section>
    <router-link class="button" to="/">Go back</router-link>
    <h1>Under construction, come back soon ;)</h1>
  </section>
</template>

<script>
export default {
  name: 'Catalog',
  data () {
    return {
      templates: []
    }
  }
}
</script>

<style scoped>
</style>
