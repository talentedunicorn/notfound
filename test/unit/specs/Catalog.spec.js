import {shallowMount} from '@vue/test-utils'
import Catalog from '@/components/Catalog'



jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({data: [ { title: "Test", slug: "test" } ]}))
}))

import axios from 'axios'

describe('Catalog.vue', () => {

  it('should load templates props', () => {
    const wrapper = shallowMount(Catalog, {
      stubs: ['router-link']
    })

    expect(axios.get).toBeCalledWith('/api/templates')
    console.log(wrapper.vm.templates)
  })

  it('should match snapshot', () => {
    const wrapper = shallowMount(Catalog, {
      stubs: ['router-link']
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})


