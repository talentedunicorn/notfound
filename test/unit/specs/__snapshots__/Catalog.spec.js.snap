// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog.vue should match snapshot 1`] = `
<section>
  <!---->
  <h1>Select template</h1>
  <p class="placeholder">No templates found</p>
</section>
`;
